app/controllers/application_controller.rb

```diff
@@ -111,6 +111,7 @@ def use_collection(id)
   def clear_session
     store_oauth_token()
     session[:user_id] = nil
+    session[:user_name] = nil
```

Comment:
In addition to :user_id, also store/clear user_name for easy interaction with URS.